### PR TITLE
Add the logic to the Calculator component.

### PR DESCRIPTION
**Pull request by petudeveloper**
For this pull request:

- Create `logic` directory.
- Add `calculate.js` and `operate.js` files.
- Implement the event handlers to use calculate.js and operate.js in a React component.
- Use state and props, pass props to child components, use lifecycle methods, lift state up.

### CODE REVIEW

File: src/components/Calculator.js [hunk 1]
```diff
@@ -1,18 +1,44 @@
...
+    ));
 
-    const utils = utilsArray.map((utilLabel, index) => <div id={`util-${index}`} key={`${utilLabel}`} className="util">{utilLabel}</div>);
-    const digits = digitsArray.map((digitLabel, index) => <div key={`${digitLabel}`} id={`digit-${index}`} className="digit">{digitLabel}</div>);
-    const operators = operatorsArray.map((operatorLabel, index) => <div id={`operator-${index}`} key={`${operatorLabel}`} className="operator">{operatorLabel}</div>);
+    const operators = operatorsArray.map((operatorLabel) => (
+      <button type="button" key={`${operatorLabel}`} className="operator" onClick={this.clickAction}>
+        {operatorLabel}
+      </button>
+    ));
```
Comment (eliasecasta) [inline src/components/Calculator.js:38]: - [x] Your division operation is not working outputting this error: image please take a look at it and fix it to comply with the project requirements 👍

**Review by eliasecasta — CHANGES_REQUESTED** [review]
# Status:  Changes Required ♻️
_Check the comments under the review_
<hr>

Hello @petumazo, I am Elias, your reviewer for this PR

Well done so far with your project!

Almost everything is working as expected, your pull request description is descriptive enough and almost all the calculator features are performing accordingly, great job!

I will mention to you a small summary of things that need to be changed in order to keep moving:

- [x] Your calculator is crashing when I try to use the division operator, please fix this to comply with the project requirements. All other operations seem to be working as expected.

Please correct those issues and send another review request, good luck, and well done so far! 

Code review by Elias Castañeda

_(Feel free to leave any questions or comments in the PR thread if something is not 100% clear)_

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


As described in the [Code reviews limit Policy](https://microverse.zendesk.com/hc/en-us/restricted?return_to=https%3A%2F%2Fmicroverse.zendesk.com%2Fhc%2Fen-us%2Farticles%2F1500004088561) you have a limited number of code reviews per project (you can check the exact number in your Dashboard). If you think that the code review was not fair, you can [request a second opinion using this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/components/Calculator.js (hunk 1 above)
Comment (petudeveloper) [reply src/components/Calculator.js:38]: Thanks for the feedback. I corrected the bug with the division operation.

**Review by BrianSammit — APPROVED** [review]
# STATUS: APROVED :green_circle: :clap: 

Hello @petumazo, 
Your project is complete!  There is nothing else to say other than... Great job 👏 
Congratulations! 🎉 

Now you can merge this branch and move on :1st_place_medal: 
HAPPY CODING AND GOOD LUCK WITH THE COMMING ONES :crossed_fingers: :four_leaf_clover: 

Well-Done-Applause


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**